From: Chris Lamb <lamby@debian.org>
Date: Tue Oct 16 12:05:35 2012 +0100
Subject: 0001-drop-supplemental-groups

---
 gunicorn/util.py | 9 +++++++++
 1 file changed, 9 insertions(+)

--- a/gunicorn/util.py
+++ b/gunicorn/util.py
@@ -4,6 +4,7 @@
 # See the NOTICE for more information.
 
 
+import errno
 import email.utils
 import fcntl
 import io
@@ -152,6 +153,14 @@
         # groups like on osx or fedora
         gid = abs(gid) & 0x7FFFFFFF
         os.setgid(gid)
+
+        # Also drop supplemental groups
+        try:
+            os.setgroups([])
+        except OSError as e:
+            if e.errno != errno.EPERM:
+                raise
+
     if uid:
         os.setuid(uid)
 
